napari/utils/events/evented_model.py

```diff
@@ -237,6 +237,33 @@ def __setattr__(self, name: str, value: Any) -> None:
             for dep in self.__field_dependents__.get(name, {}):
                 getattr(self.events, dep)(value=getattr(self, dep))
 
+    def __getattribute__(self, name: str) -> None:
+        attr = super().__getattribute__(name)
+        # avoid recursion
+        if (
+            name == '__property_setters__'
+            or name not in self.__property_setters__
+        ):
+            return attr
+
+        # wrap objects returned by properties in evented objects
+        prop = self.__property_setters__[name]
+        ret_annotation = prop.fget.__annotations__.get(
+            'return', lambda x: None
+        )
+        ret_type = get_origin(ret_annotation) or ret_annotation
+        evented_attr = ret_type(attr)
```

Comment:
this is starting to get a bit risky... is there a reason we can't just require the property itself to create an evented object?  (i.e.  why do this automagically instead of explicitly)?  

will look at the other comment about behavior soon

Reply:
No; I was trying to stay as close as I could to the behaviour of pydantic fields, while non introducing extra stuff that needs to be done to buy in. Alternatively, we could do soemthing like:

```
class A(EventedModel):
     x: ...
     _z: EventedList

    @property
    def z...
```

But I guess it's just the same as explicitly creating an evented object in the getter!

Reply:
Yeah, also note that if you did it automatically, you'd need to explicitly "type: ignore" the return values everywhere, since mypy would yell at you for returning a list when you said you would be returning an Evented object